File: 백준/Silver/10773. 제로/제로.java
import java.util.Scanner;
import java.util.Stack;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int k = sc.nextInt();
		Stack<Integer> s = new Stack<>();
		int sum = 0;

		for (int i = 0; i < k; i++) {
			int input = sc.nextInt();

			if (s.isEmpty()) {
				s.push(input);
				continue;
			}
				
			if (input == 0)
				s.pop();
			else s.push(input);
		}

		while (!s.isEmpty()) {
			sum += s.pop();
		}
		
		System.out.println(sum);

	}

}

File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <stack>

using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(nullptr);

	int K;
	cin >> K;

	stack<int> stack;
	for (int k = 0; k < K; k++)
	{
		int x;
		cin >> x;

		if (x == 0)
		{
			stack.pop();
		}
		else
		{
			stack.push(x);
		}
	}

	int sum = 0;
	while (!stack.empty())
	{
		sum += stack.top(); stack.pop();
	}

	cout << sum;
}